src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1867,7 +1868,17 @@ BuildRoutesForInsert(Query *query, DeferredErrorMessage **planningError)
 		modifyRoute = palloc(sizeof(ModifyRoute));
 
 		modifyRoute->shardId = shardInterval->shardId;
-		modifyRoute->rowValuesLists = NIL;
```

Comment:
If it's now an invariant that all `modifyRoutes` have a non-`NULL` `rowValuesList`, can you add an `Assert` somewhere to ensure we catch malformed values correctly?